engine/client/client.go

```diff
@@ -237,21 +243,33 @@ func (c *Client) startEngine(ctx context.Context) (rerr error) {
 		return err
 	}
 
+	c.connector = connector
+
 	ctx, span := Tracer().Start(ctx, "connecting to engine")
 	defer telemetry.End(span, func() error { return rerr })
 
+	logs, slog := slog.SpanLogger(ctx, InstrumentationLibrary, slog.LevelDebug)
+	defer logs.Close()
+
+	slog.Info("connecting", "runner", c.RunnerHost, "client", c.ID)
+
 	bkClient, bkInfo, err := newBuildkitClient(ctx, remote, connector)
 	if err != nil {
 		return fmt.Errorf("new client: %w", err)
 	}
 	c.bkClient = bkClient
 
 	if err := retry(ctx, 10*time.Millisecond, func(elapsed time.Duration, ctx context.Context) error {
+		slog.Info("dialing telemetry", "remote", c.RunnerHost)
```

Comment:
Oooh I *really* like this - using the `SpanLogger` gets *significantly* better results IMO, getting all the log messages to the right place :pray: 

Some of these are a bit noisy though (this is with just `-v`):

```
✔ connect 0.6s
  ✔ connecting to engine 0.1s
  ┃ 15:18:26 INF connecting runner=docker-container://dagger-engine.dev client=p                                                                  
  ┃ 166djtditnzvem0omeqm                                                                                                                          
  ┃ 15:18:26 INF dialing telemetry remote=docker-container://dagger-engine.dev                                                                    
  ┃ 15:18:26 INF dialed telemetry elapsed=8.44µs                                                                                                  
```

Reply:
Also something seems wrong in the plain progress now:

```
1   : connect
2   :   connecting to engine
2   :   [0.0s] | 15:26:13 INF connecting runner=docker-container://dagger-engine.dev client=trurchzatdscg1loohyjplb48
2   :   [0.1s] | 15:26:13 INF dialing telemetry remote=docker-container://dagger-engine.dev
2   :   [0.1s] | 15:26:13 INF dialed telemetry elapsed=6.18µs
2   :   [0.1s] | 
2   :   [0.1s] | 
2   :   connecting to engine DONE [0.1s]
3   :   starting session
3   :   [0.2s] | 
3   :   [0.2s] | 
3   :   starting session DONE [0.2s]
1   : connect DONE [0.3s]
```

Not sure where the excess newlines are coming from. Also, the double timestamps are a *little* annoying (but I can suffer).

Reply:
Moved these down to debug level, now that it respects `--debug` (https://github.com/dagger/dagger/pull/7452#discussion_r1629658902)

Reply:
@jedevc Ah just saw the second comment - those blank lines are probably the same bug that happens in Cloud now. The new logging infra send a log record on `Close()` whose body is an empty string, with `stdio.eof=true`, to indicate that you're received the last log record. The plain format must be splitting on `\n` and not handling the case where there's no linebreak.